### PR TITLE
feat(opensearchservice): support for MultiAZWithStandBy (under feature flag)

### LINKED ISSUE #26026: Amazon OpenSearch Service: High Level Constructs For OpenSearch MultiAZWithStandBy Feature
### Describe the feature

The OpenSearch Team has recently launched MultiAZWithStandby Feature for OpenSearch Domain. [Feature Documentation Link](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/managedomains-multiaz.html#:~:text=around%20these%20configurations.-,Multi%2DAZ%20with%20Standby,-Multi%2DAZ%20with).  According to the CDK docs currently there is no high level CDK construct for this feature.[CDK Docs link for the MultiAZWIthStandbyFeature](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_opensearchservice.CfnDomain.ClusterConfigProperty.html#:~:text=OpenSearch%20Service%20.-,multiAzWithStandbyEnabled%3F,-Type%3A%20boolean)

### Use Case

The general recommendation is to use high level constructs and due to lack of High level construct for this feature, the CDK template needs to be migrated to use CFN constructs if there is a need to Create an OpenSearchDomain with MultiAZWithStandBy. This serves as a hinderance to the adoption of the MultiAZWithStandBy feature for AWS OpenSearch Domains.

### Proposed Solution

Add the MultiAZWithStandByAttribute to the capacityConfig object [CDK Doc for CapacityConfig Attribute](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_opensearchservice.CapacityConfig.html) . 
 A MultiAZWithStandBy based OpenSearchDomain created using CDK high level constructs should look like
const domain = new Domain(this, 'Domain', {
  version: EngineVersion.OPENSEARCH_1_0,
  capacity: {
    masterNodes: 3,
    **multiAZWithStandbyEnabled: true (possible values: [true,false])**
  },
});

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.84.0

### Environment details (OS name and version, etc.)

macOS Ventura 13.4

### PR DESCRIPTION
**Pull request by lpizzinidev**
This fix adds support for the [`MultiAZWithStandbyEnabled`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-opensearchservice-domain-clusterconfig.html#:~:text=%3A%20No%20interruption-,MultiAZWithStandbyEnabled,Update%20requires%3A%20No%20interruption,-WarmCount) flag to the [`CapacityConfig`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_opensearchservice.CapacityConfig.html) interface.

If enabled, the `ENABLE_OPENSEARCH_MULTIAZ_WITH_STANDBY` feature flag set the default value of `MultiAZWithStandbyEnabled` to `true`

Closes #26026.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by lpizzinidev** [comment]
Exemption Request.
Let me know if I should make changes to integration tests.

File: packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts [hunk 1]
```diff
@@ -1556,6 +1566,7 @@ export class Domain extends DomainBase implements IDomain, ec2.IConnectable {
           : undefined,
         instanceCount,
         instanceType,
+        multiAzWithStandbyEnabled,
```
Comment (corymhall) [inline packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts:1569]:
```suggestion
        multiAzWithStandbyEnabled: props.capacity?.multiAzWithStandbyEnabled,
```

I don't think there is any reason to explicitly set this to false.

**Review by corymhall — CHANGES_REQUESTED** [review]
We do need an integration test for this to ensure that we can create a domain
with this enabled (I don't think it will create currently).

I think we will also need a README entry that covers the requirements and
migration. Ideally we probably want a feature flag that will allow us to set
this to true by default (unless there is a reason that this shouldn't be the
default).

File: packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts (hunk 1 above)
Comment (lpizzinidev) [reply packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts:1575]: I've added a feature flag that defaults `multiAzWithStandbyEnabled` to `true`).
This is causing some integration tests to fail.
Let me know if the implementation is correct so I can proceed to fix the failing tests.

**Comment by AmanRajAWS** [comment]
@lpizzinidev hey what are the current issues that you are blocked upon please let me know if you need any help with getting unblocked

**Review by mrgrain — DISMISSED** [review]

**Review by mrgrain — APPROVED** [review]

**Pull request merged**